internal/pally/counter.go

```diff
@@ -0,0 +1,91 @@
+package pally
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/uber-go/tally"
+)
+
+type counter struct {
+	value
+	last  int64
+	tally tally.Counter
+}
+
+func newCounter(opts Opts) *counter {
+	return &counter{value: newValue(opts)}
+}
+
+func (c *counter) diff() int64 {
+	cur := c.Load()
+	diff := cur - c.last
+	c.last = cur
+	return diff
```

Comment:
is there no synchronization issue here? is this guaranteed to be never be called concurrently?